### DISCUSSION TITLE
Native Compiler Error - Spacemacs on Emacs 29-3

### DISCUSSION THREAD

**Discussion by igorium3345** [syl20bnr/spacemacs#16396] [Q&A] [answered]
Hello all,
First of all, I apologize if this question was asked in the past. I am trying to setup Spacemacs using the latest "Emacs 29-3" version. First, I installed the standard version and I overwrote it with "Emacs 29-3 nodeps". After cloning the spacemacs files into "HOME/.emacs.d" and starting the configuration, I ran into the error below. I am new to emacs and couldn't find a solution for it. Could someone show me how to fix this ? (I am uploading a screenshot that has the most of the error lines :D) Thanks in advance.

OS: Windows 11 Home
Emacs: 29-3
Spacemacs-branch: "develop" branch

![native_compiler_error](https://github.com/syl20bnr/spacemacs/assets/108489681/fefe88bb-2d5a-4752-bccc-e6b190d2f9cf)

Debugger entered--Lisp error: (native-compiler-error (lambda (&optional arg6 arg7) (let ((f #'kill-emacs)) (funcall f arg6 arg7))) "Compiling c:/Emacs/.emacs.d/eln-cache/29.3-5efa495...")
  signal(native-compiler-error ((lambda (&optional arg6 arg7) (let ((f #'kill-emacs)) (funcall f arg6 arg7))) "Compiling c:/Emacs/.emacs.d/eln-cache/29.3-5efa495..."))
  comp--native-compile((lambda (&optional arg6 arg7) (let ((f #'kill-emacs)) (funcall f arg6 arg7))) nil "c:/Emacs/.emacs.d/eln-cache/29.3-5efa4952/subr--tr...")
  comp-trampoline-compile(kill-emacs)
  comp-subr-trampoline-install(kill-emacs)
  advice--add-function(:around (#f(compiled-function () #<bytecode 0x3c5d8019df7e91>) . #f(compiled-function (gv--val) #<bytecode 0x8fdecbbba7cb3c2>)) kill-emacs@spacemacs-really-exit nil)
  advice-add(kill-emacs :around kill-emacs@spacemacs-really-exit)
  (prog1 (defalias 'kill-emacs@spacemacs-really-exit #'(lambda (f &rest args) "Do not actually kill Emacs if a persistent server ..." (if (and (not spacemacs-really-kill-emacs) (spacemacs//persistent-server-running-p)) (spacemacs/frame-killer) (apply f args)))) (advice-add 'kill-emacs :around #'kill-emacs@spacemacs-really-exit))
  load-with-code-conversion("c:/Emacs/.emacs.d/layers/+spacemacs/spacemacs-defa..." "c:/Emacs/.emacs.d/layers/+spacemacs/spacemacs-defa..." t nil)
  load("c:/Emacs/.emacs.d/layers/+spacemacs/spacemacs-defa..." t nil)
  configuration-layer/load-file("c:/Emacs/.emacs.d/layers/+spacemacs/spacemacs-defa..." t)
  (let ((file (concat (eieio-oref obj ':dir) file))) (configuration-layer/load-file file t))
  (let ((file (car tail))) (let ((file (concat (eieio-oref obj ':dir) file))) (configuration-layer/load-file file t)) (setq tail (cdr tail)))
  (while tail (let ((file (car tail))) (let ((file (concat (eieio-oref obj ':dir) file))) (configuration-layer/load-file file t)) (setq tail (cdr tail))))
  (let ((tail files)) (while tail (let ((file (car tail))) (let ((file (concat (eieio-oref obj ...) file))) (configuration-layer/load-file file t)) (setq tail (cdr tail)))))
  (progn (let ((tail files)) (while tail (let ((file (car tail))) (let ((file (concat ... file))) (configuration-layer/load-file file t)) (setq tail (cdr tail))))))
  (if obj (progn (let ((tail files)) (while tail (let ((file (car tail))) (let ((file ...)) (configuration-layer/load-file file t)) (setq tail (cdr tail)))))))
  (let ((obj (configuration-layer/get-layer layer-name))) (if obj (progn (let ((tail files)) (while tail (let ((file ...)) (let (...) (configuration-layer/load-file file t)) (setq tail (cdr tail))))))))
  configuration-layer//load-layer-files(spacemacs-defaults ("funcs"))
  (let ((layer-name (car tail))) (configuration-layer//load-layer-files layer-name files) (setq tail (cdr tail)))
  (while tail (let ((layer-name (car tail))) (configuration-layer//load-layer-files layer-name files) (setq tail (cdr tail))))
  (let ((tail layer-names)) (while tail (let ((layer-name (car tail))) (configuration-layer//load-layer-files layer-name files) (setq tail (cdr tail)))))
  configuration-layer//load-layers-files((spacemacs-bootstrap spacemacs-visual spacemacs-purpose spacemacs-project spacemacs-org spacemacs-navigation spacemacs-modeline spacemacs-misc spacemacs-language spacemacs-evil spacemacs-editing-visual spacemacs-editing spacemacs-layouts spacemacs-completion spacemacs-defaults spacemacs-base spacemacs emacs-lisp helm multiple-cursors treemacs) ("funcs"))
  configuration-layer//load()
  (cond (changed-since-last-dump-p (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message "Layer list has changed since last dump.") (configuration-layer//dump-emacs)))) (spacemacs-force-dump (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "--force-dump passed on the command line or configu..." "been reloaded, forcing a redump.")) (configuration-layer//dump-emacs)))) ((spacemacs-is-dumping-p) (configuration-layer//load)) ((and (spacemacs/emacs-with-pdumper-set-p) (spacemacs-run-from-dump-p)) (configuration-layer/message "Running from a dumped file. Skipping the loading p...")) (t (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "Layer list has not changed since last time. " "Skipping dumping process!"))))))
  (let (changed-since-last-dump-p) (if (file-exists-p configuration-layer--last-dotspacemacs-configuration-layers-file) (progn (configuration-layer/load-file configuration-layer--last-dotspacemacs-configuration-layers-file))) (let ((layers dotspacemacs-configuration-layers)) (progn (if "Calling dotfile layers..." (progn (spacemacs-buffer/message "Calling dotfile layers..."))) (if (fboundp 'dotspacemacs/layers) (progn (condition-case err (dotspacemacs/layers) ((debug error) (configuration-layer//increment-error-count) (spacemacs-buffer/append ... t)))))) (setq dotspacemacs--configuration-layers-saved dotspacemacs-configuration-layers) (setq changed-since-last-dump-p (not (equal layers dotspacemacs-configuration-layers))) (spacemacs/dump-vars-to-file '(dotspacemacs-configuration-layers) configuration-layer--last-dotspacemacs-configuration-layers-file)) (cond (changed-since-last-dump-p (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message "Layer list has changed since last dump.") (configuration-layer//dump-emacs)))) (spacemacs-force-dump (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "--force-dump passed on the command line or configu..." "been reloaded, forcing a redump.")) (configuration-layer//dump-emacs)))) ((spacemacs-is-dumping-p) (configuration-layer//load)) ((and (spacemacs/emacs-with-pdumper-set-p) (spacemacs-run-from-dump-p)) (configuration-layer/message "Running from a dumped file. Skipping the loading p...")) (t (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "Layer list has not changed since last time. " "Skipping dumping process!")))))))
  configuration-layer/load()
  (let ((please-do-not-disable-file-name-handler-alist nil)) (require 'core-spacemacs) (spacemacs/dump-restore-load-path) (configuration-layer/load-lock-file) (spacemacs/init) (configuration-layer/stable-elpa-init) (configuration-layer/load) (spacemacs-buffer/display-startup-note) (spacemacs/setup-startup-hook) (spacemacs/dump-eval-delayed-functions) (if (and dotspacemacs-enable-server (not (spacemacs-is-dumping-p))) (progn (require 'server) (if dotspacemacs-server-socket-dir (progn (setq server-socket-dir dotspacemacs-server-socket-dir))) (if (server-running-p) nil (message "Starting a server...") (server-start)))))
  (if (not (version<= spacemacs-emacs-min-version emacs-version)) (error (concat "Your version of Emacs (%s) is too old. " "Spacemacs requires Emacs version %s or above.") emacs-version spacemacs-emacs-min-version) (let ((please-do-not-disable-file-name-handler-alist nil)) (require 'core-spacemacs) (spacemacs/dump-restore-load-path) (configuration-layer/load-lock-file) (spacemacs/init) (configuration-layer/stable-elpa-init) (configuration-layer/load) (spacemacs-buffer/display-startup-note) (spacemacs/setup-startup-hook) (spacemacs/dump-eval-delayed-functions) (if (and dotspacemacs-enable-server (not (spacemacs-is-dumping-p))) (progn (require 'server) (if dotspacemacs-server-socket-dir (progn (setq server-socket-dir dotspacemacs-server-socket-dir))) (if (server-running-p) nil (message "Starting a server...") (server-start))))))
  load-with-code-conversion("c:/Emacs/.emacs.d/init.el" "c:/Emacs/.emacs.d/init.el" t t)
  load("c:/Emacs/.emacs.d/init" noerror nomessage)
  startup--load-user-init-file(#f(compiled-function () #<bytecode -0x1cae2558185f4e66>) #f(compiled-function () #<bytecode -0x1f3c61addc0c4f35>) t)
  command-line()
  normal-top-level()

**Comment by smile13241324** [comment, marked as answer]
I am sorry but we do not longer support native windows builds as these create lot's of subtle issues in emacs especially with complex features like native compilation.

Therefore please move to wsl2 with wslg instead. Running emacs directly in wsl2 will open it via X or wayland (depending on which emacs build you have installed) and run just like any other windows application.

Here are some instructions how to install wsl2 https://learn.microsoft.com/en-us/windows/wsl/install.
